server/src/modules/reports/reports.service.ts

```diff
@@ -1,19 +1,28 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Inject } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { ReturnModelType } from '@typegoose/typegoose';
 import { ObjectIdScalar } from '../common/graphql-scalars/object-id.scalar';
 import { AddReportInput } from './models/add-report.input';
 import { EditReportInput } from './models/edit-report.input';
 import { Report } from './models/report.schema';
 import { ICurrentUser } from '../auth/interfaces/current-user.interface';
+import { User } from '../users/models/user.schema';
+import { UsersService } from '../users/users.service';
+import { ObjectId } from 'bson';
 
 @Injectable()
 export class ReportsService {
+
   constructor(
     @InjectModel(Report.name)
     private readonly reportModel: ReturnModelType<typeof Report>,
+    @Inject(UsersService) private readonly usersService: UsersService,
   ) {}
 
+  async resolveUser(reportedBy: any): Promise<User> {
+   return this.usersService.findOne(reportedBy);
```

Comment:
Check if `reportedBy` is `User` type, if so, return that user, else find user in db.